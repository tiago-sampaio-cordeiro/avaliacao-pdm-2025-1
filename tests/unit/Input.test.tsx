import { render, fireEvent, screen } from "@testing-library/react-native";
import CustomInput from "../../components/Input";

describe("CustomInput", () => {
  it("deve renderizar o label e o placeholder corretamente", () => {
    render(
      <CustomInput
        label="Nome"
        value=""
        onChangeText={() => {}}
        placeholder="Digite seu nome"
      />
    );

    expect(screen.getByText("Nome")).toBeTruthy();
    expect(screen.getByPlaceholderText("Digite seu nome")).toBeTruthy();
  });

  it("deve chamar onChangeText quando o texto muda", () => {
    const mockFn = jest.fn();

    render(
      <CustomInput
        label="Email"
        value=""
        onChangeText={mockFn}
        placeholder="Digite seu email"
      />
    );

    const input = screen.getByPlaceholderText("Digite seu email");
    fireEvent.changeText(input, "teste@example.com");

    expect(mockFn).toHaveBeenCalledWith("teste@example.com");
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it("deve bater com o snapshot atual", () => {
    const tree = render(
      <CustomInput
        label="Senha"
        value=""
        onChangeText={() => {}}
        placeholder="Digite sua senha"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
