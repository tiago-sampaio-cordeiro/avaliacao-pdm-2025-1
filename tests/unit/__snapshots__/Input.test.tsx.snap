// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput deve bater com o snapshot atual 1`] = `
<View
  style={
    {
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "600",
        "marginBottom": 6,
      }
    }
  >
    Senha
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Digite sua senha"
    placeholderTextColor="#aaa"
    style={
      {
        "backgroundColor": "#f0f0f0",
        "borderRadius": 8,
        "fontSize": 16,
        "padding": 12,
      }
    }
    value=""
  />
</View>
`;
